www/js/playerlib.js

```diff
@@ -1854,7 +1876,7 @@ function filterLib() {
 			allArtists.sort(function(a, b) {
 				return collator.compare(removeArticles(a), removeArticles(b));
 			});
-			allAlbumsTmp.sort(function(a, b) {
+			allAlbums.sort(function(a, b) {
 				return collator.compare(removeArticles(a['album']), removeArticles(b['album']));
```

Comment:
@moodeaudio I noticed that `removeArticles` only removes english articles. Wouldn't it make sense to use `album_sort` or `artist_sort` tags instead?

Reply:
Are those tags normally populated in tracks or by ripping software?

Reply:
Not as far as I know. However it would give the user back control of how they want their library sorted. And these tags are created for specifically this use-case. It would avoid an infinity of english language centric bugs being reported. (eg: how come we sort with `Beatles` instead of `The Beatles` but not `Tigre` instead of `Le Tigre`)

Reply:
IMO, If those tags are populated, then we should use them. Perhaps we could fallback to removing english only articles.  This also opens the possibility for users to sort artists `lastname, firstname` or other such OCD schemes.

Reply:
It makes sense to use them if populated. We just need a way to let the client know whether or not to use removeArticles().

Reply:
moOde 5 includes a setting on the Library Options screen to allow user to define the list of articles to be ignored or if user blanks out the list then no articles would be ignored. 

I also included logic in function genLibrary($flat) to substitute ArtistSort and AlbumSort tag values for the default tag values Artist/AlbumArtist and Album.  Apparently this is bad. Users posted in moOde Forum that only part of their collection had the sort tags populated and this really messed up the sorts in the Library views. Partial population occurs when users have used various tagging utils on their collection. Some utils populate the tags (Picard) and other don't. 

Other comments were that the sort tags should only be used for sorting and that the default tags should be used for display. 

That makes sense and this could certainly be done but it would only work if the entire collection had the Sort tags populated. If the collection had partially populated tags it would turn out bad.

Id like to use these tags but I think it would require a setting with a help that explained that ALL the sort tags have to be populated for this to work.

-Tim

Reply:
@moodeaudio It's not quite clear why partially tagged collection wouldn't sort correctly. `allAlbums.sort((a,b) => (a.sortArtist ||  a.artist).localeCompare(b.sortArtist || b.artist))`

Reply:
I agree that the sortArtist and the sortAlbum tags should only be used for sorting.  The displayed name should always be artist or album.

Reply:
When only some of the sort tags are populated we can end up with something like below

Artist = Tim Curtis | ArtistSort = <does not exist>' | Album = Blue Sky
Artist = Tim Curtis | ArtistSort = Curtis, Tim | Album = Red River

Artist = exists(ArtistSort) ? ArtistSort : Artist

In this case the artist Tim Curtis will end up appearing twice in the artist list, once under the T's and again under the C's

Reply:
@moodeaudio This seems correct given inconsistent tagging.